sdks/java/io/rabbitmq/src/test/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIOTest.java

```diff
@@ -0,0 +1,127 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.rabbitmq;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.QueueingConsumer;
+
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.qpid.server.Broker;
+import org.apache.qpid.server.BrokerOptions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * Test of {@link RabbitMqIO}.
+ */
+public class RabbitMqIOTest {
+
+  public static final int PORT = 5672;
+
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  private Broker broker;
+
+  @Before
+  public void startBroker() throws Exception {
+    broker = new Broker();
+    BrokerOptions options = new BrokerOptions();
+    options.setConfigProperty("qpid.amqp_port", String.valueOf(PORT));
+    options.setConfigProperty("qpid.work_dir", Files.createTempDirectory("qpid").toString());
+    options.setConfigProperty("qpid.home_dir", "src/test/qpid");
+    broker.startup(options);
+  }
+
+  @After
+  public void stopBroker() throws Exception {
+    broker.shutdown();
+  }
+
+  @Test
+  public void testRead() throws Exception {
+    PCollection<byte[]> output = pipeline.apply(
+        RabbitMqIO.read().withUri("amqp://guest:guest@localhost:" + PORT).withQueue("READ")
+            .withMaxNumRecords(10));
+    PAssert.that(output)
+        .containsInAnyOrder("Test 0".getBytes(), "Test 1".getBytes(), "Test 2".getBytes(),
+            "Test 3".getBytes(), "Test 4".getBytes(), "Test 5".getBytes(), "Test 6".getBytes(),
+            "Test 7".getBytes(), "Test 8".getBytes(), "Test 9".getBytes());
+
+    ConnectionFactory connectionFactory = new ConnectionFactory();
+    connectionFactory.setUri("amqp://guest:guest@localhost:" + PORT);
+    Connection connection = connectionFactory.newConnection();
+    Channel channel = connection.createChannel();
+    channel.queueDeclare("READ", false, false, false, null);
```

Comment:
Here too, the publisher should not be declaring queues, and rather it should be publishing to an exchange.